### ISSUE TITLE
Mapping null values for nested fields does not work

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#94]
```
What steps will reproduce the problem?
1. Register a mapping for a nested property
2. Set mapNulls to true at the global level or at the field level 
3. Try mapping a null value from the source object to the destination one
4. Run the attached JUnit test

What is the expected output? What do you see instead?
- The expected behavior would be to map the null value from the source object 
to the nested property of the destination object.
Instead the JUnit fails with java.lang.AssertionError: EndDate is not null


What version of the product are you using? On what operating system?
- Orika 1.4.1
- Java 1.7 
- OS: Windows 7 Professional


Please provide any additional information below.

```

Original issue reported on code.google.com by `elenamar...@gmail.com` on 8 Apr 2013 at 10:36

Attachments:
- [OrikaTestNulls.java](https://storage.googleapis.com/google-code-attachments/orika/issue-94/comment-0/OrikaTestNulls.java)

**Comment by GoogleCodeExporter** [comment]
```
Thanks for reporting the issue.
We will get a closer look.
```

Original comment by `elaat...@gmail.com` on 9 Apr 2013 at 1:11
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
I'll take this one...
```

Original comment by `matt.deb...@gmail.com` on 17 Apr 2013 at 5:35

**Comment by GoogleCodeExporter** [comment]
```
A fix is available in the current master, and should make it into 1.4.2 release.
Just a few notes on your test case:
(1) You had correct setup for mapping nulls at the global level, so I'll skip 
that one.
(2) To map nulls at the class-map level, you need specify mapNulls(true) before 
adding other field maps, as they will take on whatever is configured for the 
ClassMap as their default, like this:
        classMapBuilder.mapNulls(true);
        classMapBuilder.field("startDate", "startDate");
        classMapBuilder.field("endDate", "inner.endDate");
(3) Finally, to map nulls at a field-level, you need to use:
        classMapBuilder.fieldMap("startDate", "startDate").mapNulls(true).add();
      (INSTEAD OF) 
        classMapBuilder.field("startDate", "startDate").mapNulls(true);
   ... the second one is actually calling mapNulls on the ClassMapBuilder, not on the field-map just added (and it was calling it after the field map had already been created, so it only affects field maps added after it);
I know it's not completely clear, especially since you can call mapNulls on the 
classMap and the fieldMap...maybe we can look into an alternate (additional) 
signature that makes it more intuitive


```

Original comment by `matt.deb...@gmail.com` on 17 Apr 2013 at 8:09

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.deb...@gmail.com` on 17 Apr 2013 at 8:40
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
I did ran the OrikaTestNulls.java  Junit on 1.4.1-SNAPSHOT by using the mapping 
at a global level but also on field level as shown on (3) on the 3rd comment. 

Unfortunately the endDate is still not null and the JUnit fails. 
Am i maybe using the wrong version ? 
```

Original comment by `BinaryX...@gmail.com` on 29 Apr 2013 at 9:04

**Comment by GoogleCodeExporter** [comment]
```
You want the 1.4.2-SNAPSHOT version
```

Original comment by `matt.deb...@gmail.com` on 29 Apr 2013 at 6:30

**Comment by GoogleCodeExporter** [comment]
```
Both our implementation and JUnit seem to work well. 
Thank you 
```

Original comment by `BinaryX...@gmail.com` on 30 Apr 2013 at 12:07

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.4.2 release
```

Original comment by `matt.deb...@gmail.com` on 20 May 2013 at 6:41
- Changed state: **Fixed**
